test/lib/ansible_test/_data/pytest.ini

```
@@ -1,3 +1,4 @@
 [pytest]
 xfail_strict = true
 mock_use_standalone_module = true
+junit_family = legacy
```

@webknjaz I'd prefer to use `xunit1` since it's more specific, but equivalent to `legacy`:

```suggestion
junit_family = xunit1
```

That being said, I'm curious what failures will occur without this. I'm only aware of the deprecation warnings.

Also, it's worth pointing out that there are actually at least five output formats, not two:

- Python 2.6 (pytest 3.2.5)
- Python 2.7 (pytest 4.6.6) xunit1
- Python 2.7 (pytest 4.6.6) xunit2
- Python 3.5+ (pytest 5.2.1) xunit1
- Python 3.5+ (pytest 5.2.1) xunit2

---

`xunit2` appeared in pytest in v4.2.0. So it should be good for all cases except for Python 2.6. Can we set it in the config and then conditionally pass `xunit1` via args on invocation?

---

According to https://github.com/pytest-dev/pytest/blob/3.2.5/_pytest/junitxml.py vs https://github.com/pytest-dev/pytest/blob/4.2.0/src/_pytest/junitxml.py, pytest v3.2.5 doesn't process `junit_family` arg at all.

So Python 2.6 will completely ignore whatever we set there and Python 2.7+ will respect it and all pytest versions that we use understand `xunit2`.

I'd say just set `xunit2` and that's it.

---

I don't think we should change the format we're using at this point, which means we would need to either use `legacy` or `xunit1` -- but since those two are identical, I'd prefer using `xunit1` since it's more clear what format will be used.

---

Is this related to the parser used on the other end?

---

Basically, yes. I'd like to avoid making changes now that could potentially break existing setups that are parsing the output. Using `xunit1` will maintain backwards compatibility with existing systems.

Perhaps later we can look at using a newer format, but that's not something I want to change now.

---

Fair enough